Fix outdated option name

blm768 opened the pull request:
While muddling through #381, I realized the docs had a reference to an option that had been renamed.

I'm not sure yet whether any changes should be made to the example in the previous section, as the default seems to be to pull stuff in from `hmUsers` rather than directly specifying options like the example does, but I'm not sure what we'd consider best practice there.

Pacman99 commented:
Wow I'm pretty sure that option has always been named `home`, I can't believe I missed that.
Thanks for catching this!

Pacman99 commented:
Ohh I see I think we need be more clear as to what we're referring to, but the option is actually `home-manager`. That code is using the [home-manager nixos module](https://github.com/nix-community/home-manager/blob/master/nixos/default.nix), not the digga api.

blm768 commented:
> That code is using the home-manager nixos module, not the digga api.

Right. I think the example should work as given, but it doesn't match the current state of the template, which presumably expects the profiles to be configured in `flake.nix`:

```nix
{
  home-manager.users = { inherit (hmUsers) nixos; };
}
```

blm768 commented:
> Ohh I see I think we need be more clear as to what we're referring to, but the option is actually `home-manager`. That code is using the [home-manager nixos module](https://github.com/nix-community/home-manager/blob/master/nixos/default.nix), not the digga api.

Were you referring to the option I changed or the option in the example above it? It's been a while now, but I think I remember that the one I changed was using `digga`'s API.

Pacman99 commented on doc/concepts/users.md:
```diff
@@ -39,7 +39,7 @@ argument that gets passed to your home-manager users.
 ### Example
```
```suggestion
### Example
`flake.nix`:
```

Pacman99 commented:
Right yeah you were correct originally, I'm sorry about my confusion.
I just added a comment with a slight change to specify what file the docs are talking about. With that, I'm happy to merge this.
